modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/config/XmlExtensionParserUtils.java

```diff
@@ -307,6 +312,11 @@ static ResolverSet getResolverSet(ElementDescriptor element, List<ParameterModel
                     resolver = new StaticValueResolver(null);
                 }
 
+                if (parameterModel.getExpressionSupport() == ExpressionSupport.LITERAL)
```

Comment:
this branch evaluates many conditions, but only one of them should be true for a given parameter model. That means that once you hit a possitve, you shouldn't continue evaluating the rest.

That been said, the literal case should be evaluated before the parseParameter case above